spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/FlywayEndpoint.java

```diff
@@ -38,21 +38,25 @@
  * @since 1.3.0
  */
 @ConfigurationProperties(prefix = "endpoints.flyway")
-public class FlywayEndpoint extends AbstractEndpoint<List<FlywayMigration>> {
+public class FlywayEndpoint extends AbstractEndpoint<List<List<FlywayMigration>>> {
 
-	private final Flyway flyway;
+	private final List<Flyway> flyway;
 
-	public FlywayEndpoint(Flyway flyway) {
+	public FlywayEndpoint(List<Flyway> flyway) {
 		super("flyway");
 		Assert.notNull(flyway, "Flyway must not be null");
 		this.flyway = flyway;
 	}
 
 	@Override
-	public List<FlywayMigration> invoke() {
-		List<FlywayMigration> migrations = new ArrayList<FlywayMigration>();
-		for (MigrationInfo info : this.flyway.info().all()) {
-			migrations.add(new FlywayMigration(info));
+	public List<List<FlywayMigration>> invoke() {
```

Comment:
Changing to `List<List<FlywayMigration>>` breaks backwards compatibility without any benefit. I think we should either keep to old format, or change to a Map<String, List<FlywayMigration>> where the string somehow indicates which DataSource the migrations were run against. The JDBC URL of the DB would be ideal, but I'm not sure we can get that.